protocol_rfcs/managed-commits.md

```diff
@@ -0,0 +1,315 @@
+
+## Sample Commit Owner API
+
+The following is an example of a possible commit-owner API which some Java-based Delta client might require commit-owner implementations to target:
+
+```java
+
+interface CommitStore {
+    /**
+     * Commits the given set of `actions` to the given commit `version`.
+     *
+     * @param version The version we want to commit.
+     * @param actions Actions that need to be committed.
+     *
+     * returns CommitResponse which has details around the new committed delta file.
+     */
+    def commit(
+        version: Long,
+        actions: Iterator[String]): CommitResponse
+
+    /**
+     * API to get the un-backfilled commits for the table represented by the given `tablePath` after
+     * the given `startVersion`. The returned commits are contiguous and in ascending version order.
+     * Note that the first version returned by this API may not be equal to the `startVersion`. This
+     * happens when few versions starting from `startVersion` are already backfilled and so
+     * CommitStore may have stopped tracking them.
+     *
+     * @return a list of `Commit` which are tracked by commit-owner.
+     *
+     */
+    def getCommits(startVersion: Long): Seq[Commit]
+
+    /**
+     * API to ask the commit-owner to backfill all commits <= given `version`.
+     */
+    def backfillToVersion(version: Long): Unit
+}
+```
+
+## Commit Backfills
+Backfilling is the process of copying the un-backfilled commits i.e. `_delta_log/_commits/<version>.<uuid>.json` to `_delta_log/<version>.json`.
+With the help of backfilling, the [delta files](#delta-log-entries) are visible even to the filesystem based Delta clients that do not
+understand `managed-commits`. Backfill also allows the commit-owner to reduce the number of commits it must track internally.
+
+Backfill must be sequential. In other words, a commit-owner must ensure that backfill of commit `v-1` is complete before initiating backfill of commit 'v'.
+
+`commit-owner`s are encouraged to backfill the commits frequently. This has several advantages:
+1. Filesystem-based Delta implementations may only understand backfilled commits, and frequent backfill allows them to access the most recent table snapshots.
+2. Frequent backfilling minimizes the chance of data loss in case the `commit-owner` loses its state due to some failover/issue.
+3. Some maintenance operations (such as checkpoints, log compaction, and metadata cleanup) can be performed only on the backfilled part of the table. Refer to the [Maintenance operations on managed-commit tables](#maintenance-operations-on-managed-commit-tables) section for more details.
+
+The commit-owner may also expose an API to backfill the commits. This will allow clients to ask the commit-owner to backfill the commits if needed in order to do some maintenance operations.
```

Comment:
Rephrased this to:

> The commit-owner also needs to expose an API to backfill the commits. This will allow clients to ask the commit-owner to backfill the commits if needed in order to do some maintenance operations.